### PR TITLE
Reduce GPU mem usuage

### PR DESCRIPTION
**Pull request by sbinnee**
Using [`torch.no_grad()`] reduces GPU memory usage significantly. But I am aware that Cellpose not only uses pytorch for GPU calculation. Any idea how to detect GPU backend and take advantage of [`torch.no_grad()`] if it is pytorch?

Related issue in image.sc forum
- [`torch.no_grad()`]
- [https://forum.image.sc/t/napari-cellpose-out-of-memory/60742/3](https://forum.image.sc/t/napari-cellpose-out-of-memory/60742/3)

[`torch.no_grad()`]: https://pytorch.org/docs/stable/generated/torch.no_grad.html?highlight=no_grad#torch.no_grad

### CODE REVIEW

**Comment by tlambert03** [comment]
sorry for the general inactivity here.  I'm afraid I don't know enough to comment either way here.  Can you think of _any_ potential downsides here?  Seems like since it's never training, only inferring, it should be safe, yes?

Also, since they don't directly depend on torch in this package, we should use a `try: import` block (with a no-op fallback decorator)

**Comment by tlambert03** [comment]
let's go with it :) added a try/import block for you
